Settings.c

```diff
@@ -73,20 +73,27 @@ static void writeQuotedList(FILE* fp, char** list) {
 
 */
 
-void Settings_delete(Settings* this) {
-   free(this->filename);
-   free(this->initialFilename);
-   for (unsigned int i = 0; i < HeaderLayout_getColumns(this->hLayout); i++) {
+static void Settings_deleteColumns(Settings* this) {
+   for (size_t i = 0; i < HeaderLayout_getColumns(this->hLayout); i++) {
       String_freeArray(this->hColumns[i].names);
       free(this->hColumns[i].modes);
    }
    free(this->hColumns);
+}
+
+static void Settings_deleteScreens(Settings* this) {
    if (this->screens) {
-      for (unsigned int i = 0; this->screens[i]; i++) {
+      for (size_t i = 0; this->screens[i]; i++)
          ScreenSettings_delete(this->screens[i]);
-      }
       free(this->screens);
```

Comment:
This should set `this->screens = NULL;` explicitly, to avoid potential UAF issues.

Reply:
Nak, as above.  That kinda change would warrant more general buy-in IMO, it affects the entire code base.